### PR TITLE
feat(test): enable running very_good test --optimization for specific files/directory

### PR DESCRIPTION
**Pull request by agacemi**
## Description

In the current version, optimization is disabled if we give additional directoires or files to `very_good test` command line. The goal of this PR is to add the ability to run `'very_good test' with optimization on a specific files or directoriesc or both.  

``` bash
# Optimization will be enabled
very_good test  test/counter/cubit
very_good test  test/counter/view/counter_view_test.dart  
very_good test  test/counter/view  test/counter/cubit

```
## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: test/src/cli/flutter_cli_test.dart [hunk 1]
```diff
@@ -864,10 +864,14 @@ void main() {
...
+        final originalVars = <String, dynamic>{
+          'package-root': directory.path,
+          'test-paths': <String>[],
+        };
         final updatedVars = <String, dynamic>{
           'package-root': directory.path,
-          'foo': 'bar'
+          'foo': 'bar',
+          'test-paths': <String>[]
         };
```
Comment (renancaraujo) [inline test/src/cli/flutter_cli_test.dart:875]: Need a new test with some paths

**Review by renancaraujo — COMMENTED** [review]
I left here some remarks about the overall implementation. 

Thanks for the time you took for this. It may be better to have an issue before a PR so we can discuss and understand better on top of the problem. 

Also, your implementation seems to be breaking the e2e test (which proves that this is a breaking change), may be fas to fix (just remove the path argument). 


If you want to follow up on this, wait for a couple of days when the test_runner brick will be unbundled in this repo (so things may be easier to review and update).

**Comment by agacemi** [comment]
> I left here some remarks about the overall implementation.
> 
> Thanks for the time you took for this. It may be better to have an issue before a PR so we can discuss and understand better on top of the problem.
> 
> Also, your implementation seems to be breaking the e2e test (which proves that this is a breaking change), may be fas to fix (just remove the path argument).
> 
> If you want to follow up on this, wait for a couple of days when the test_runner brick will be unbundled in this repo (so things may be easier to review and update).

Thank you for reviewing. 

At first, I wanted to solve the issue https://github.com/VeryGoodOpenSource/very_good_cli/issues/508. On other hand, our team expressed this need : We have more than 4000 tests. Our base code is organized around many folders, each folder deals with a feature managed by a team.  Thus, we wanted to speed up tests by enabling optimization if very_good est executed on a given folder.  Hence, my PR.

Regarding the breaking change, I have not idea what could be  trailing arguments except directories or files path ?! This PR assume that the rest of result, if given, are necessarily specific files or specific directories.  Maybe I was wrong

Please, let me know when test_runner brick is unbundled to resume some fix you requested.

**Comment by renancaraujo** [comment]
> Please, let me know when test_runner brick is unbundled to resume some fix you requested.

#639 is merged so that the changes can be done to the `test_opitimizer` brick and bundled via `tool/generate_test_optimizer_bundle.sh`

**Comment by renancaraujo** [comment]
@agacemi do you plan to followup on this?

**Comment by agacemi** [comment]
> 

@renancaraujo Yes absolutely. I'am just waiting for the acceptation of merge on my other PR https://github.com/VeryGoodOpenSource/very_good_cli/pull/563 to avoid rebase hell :) and approval dismissing.

**Comment by tomarra** [comment]
@agacemi looking for an update on this PR from your side. It's been sitting for some time so it has some merge conflicts that need to be solved.

**Comment by agacemi** [comment]
> @agacemi looking for an update on this PR from your side. It's been sitting for some time so it has some merge conflicts that need to be solved.

@tomarra Absolutely, I'm aware. Currently, I can't spent time to maintain to follow this PR. for this reason, it's in draft status

**Pull request closed without merging**